pkg/datapath/linux/probes/probes.go

```diff
@@ -217,3 +223,60 @@ func (p *ProbeManager) GetHelpers(prog string) map[string]struct{} {
 	}
 	return nil
 }
+
+// writeHeaders executes bpftool to generate BPF feature C macros and then
+// writes them to the given writer.
+func (p *ProbeManager) writeHeaders(featuresFile io.Writer) error {
+	cmd := exec.WithTimeout(
+		defaults.ExecTimeout, "bpftool", "feature", "probe", "macros")
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf(
+			"could not initialize stdout pipe for bpftool: %s", err)
+	}
+	stderrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf(
+			"could not initialize stderr pipe for bpftool: %s", err)
+	}
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("could not start bpftool: %s", err)
+	}
+
+	writer := bufio.NewWriter(featuresFile)
+	defer writer.Flush()
+
+	go io.Copy(writer, stdoutPipe)
+	if err := cmd.Wait(); err != nil {
+		stderr, err := ioutil.ReadAll(stderrPipe)
+		if err != nil {
+			return fmt.Errorf(
+				"bpftool did not run succesfully and stdout could not be read: %s",
```

Comment:
Typo on `successfully` here. Maybe this could be shorten to `bpftool failed: %s` though.